ec/ecresource/deploymentresource/update.go

```diff
@@ -83,7 +92,11 @@ func (r *Resource) Update(ctx context.Context, req resource.UpdateRequest, resp
 	updatePrivateStateTrafficFilters(ctx, resp.Private, planRules)
 	resp.Diagnostics.Append(v2.HandleRemoteClusters(ctx, r.client, plan.Id.ValueString(), plan.Elasticsearch)...)
 
-	deployment, diags := r.read(ctx, plan.Id.ValueString(), &state, &plan, res.Resources, planRules)
+	var newMigrateTemplateRequest *deployments.MigrateDeploymentTemplateOK
+	deployment, diags := r.read(ctx, plan.Id.ValueString(), &state, &plan, res.Resources, planRules, newMigrateTemplateRequest)
+
+	// Store migrate request in private state
+	updatePrivateStateMigrateTemplateRequest(ctx, resp.Private, newMigrateTemplateRequest)
```

Comment:
`newMigrateTemplateRequest` is always nil here. But, anyway we can remove the line - I guess we only need to store it to cache in `Read`.

Reply:
Addressed in https://github.com/elastic/terraform-provider-ec/pull/755/commits/4c70d0034521bc6e2be803e5472948612fb64f2c.